amaranth/hdl/_ast.py

```diff
@@ -1733,8 +1734,17 @@ def __repr__(self):
                                            self.width, self.stride)
 
 
+class _CatMeta(ABCMeta):
```

Comment:
I'd rather make `Cat` a function, renaming the concatenation class to `Concat` and the IO version to `IOConcat` for disambiguation.

Reply:
Actually, do you even need a metaclass here? Wouldn't `__new__` suffice?

Reply:
Ok, going with `__new__`